Error: Cannot find module '.../src/theme.css.ts.vanilla.css!=!../../../../../../..'

<!-- NOTE: Issues should only be used for actual issues. Any feature requests or general feedback should be raised in Github Discussions (https://github.com/seek-oss/vanilla-extract/discussions) or Discord (https://discord.gg/6nCfPwwz6w) -->

### Describe the bug
<!-- A clear and concise description of what the bug is, including any error message that were displayed. -->

Hi, I've stumbled upon a bug which has been reported previously (https://github.com/seek-oss/vanilla-extract/issues/396) but the issue was closed without resolving.

I'm using Next.js with vanilla-extract and I get the following error in dev mode:

```
error - Error: Cannot find module 'organization/libs/ui/theme/src/theme.css.ts.vanilla.css!=!../../../../../../../node_modules/virtual-resource-loader/dist/virtual-resource-loader.cjs.js?{"source":"LnRoZW1lX18xdmh0NGJqMCB7CiAgLS1jb2xvcnMtZ3JheS0wX18xdmh0NGJqMTogI0ZGRkZGRjsKICAtLWNvbG9ycy1ncmF5LTNfXzF2aHQ0YmoyOiAjRjhGOEY4OwogIC0tY29sb3JzLWdyYXktNV9fMXZodDRiajM6ICNGM0YyRjQ7CiAgLS1jb2xvcnMtZ3JheS0xNV9fMXZodDRiajQ6ICNEQkRCRTA7CiAgLS1jb2xvcnMtZ3JheS0zNV9fMXZodDRiajU6ICNBQUFBQjc7CiAgLS1jb2xvcnMtZ3JheS03MF9fMXZodDRiajY6ICM1NjU1NzA7CiAgLS1jb2xvcnMtZ3JheS0xMDBfXzF2aHQ0Ymo3OiAjMEQwQjMyOwogIC0tY29sb3JzLW5hdnktbGlnaHRfXzF2aHQ0Ymo4OiAjMkEzQjkwOwogIC0tY29sb3JzLW5hdnktYmFzZV9fMXZodDRiajk6ICMyODI1NjE7CiAgLS1jb2xvcnMtYmx1ZS1zb2Z0X18xdmh0NGJqYTogI0UyRjZGRjsKICAtLWNvbG9ycy1ibHVlLWxpZ2h0X18xdmh0NGJqYjogIzk2RDVFRjsKICAtLWNvbG9ycy1ibHVlLWJhc2VfXzF2aHQ0YmpjOiAjMjZBOUUxOwogIC0tY29sb3JzLXJlZC1iYXNlX18xdmh0NGJqZDogI0RFNjc2NzsKICAtLWNvbG9ycy1ncmVlbi1saWdodF9fMXZodDRiamU6ICM2NERGQTQ7CiAgLS1jb2xvcnMtZ3JlZW4tYmFzZV9fMXZodDRiamY6ICMyQUI2NzM7CiAgLS1jb2xvcnMtZ3JhZGllbnQtZ3JlZW5fXzF2aHQ0YmpnOiBsaW5lYXItZ3JhZGllbnQoOTBkZWcsICMyQkI2NzMgMCUsICMyN0FBRTEgMTAwJSk7CiAgLS1jb2xvcnMtZ3JhZGllbnQtYmx1ZV9fMXZodDRiamg6IGxpbmVhci1ncmFkaWVudCgxODBkZWcsICMyQTNCOTAgMTMuNyUsICMzRTRDOTcgMTAwJSk7CiAgLS1mb250U2l6ZXMtaDFfXzF2aHQ0YmppOiAzNnB4OwogIC0tZm9udFNpemVzLWgyX18xdmh0NGJqajogMjRweDsKICAtLWZvbnRTaXplcy1oM19fMXZodDRiams6IDE4cHg7CiAgLS1mb250U2l6ZXMtYmFzZV9fMXZodDRiamw6IDE4cHg7CiAgLS1mb250U2l6ZXMtbGFiZWwtYmFzZV9fMXZodDRiam06IDE2cHg7CiAgLS1mb250U2l6ZXMtbGFiZWwtc21fXzF2aHQ0YmpuOiAxNHB4OwogIC0tZm9udFNpemVzLWFjY2VudC1iYXNlX18xdmh0NGJqbzogMThweDsKICAtLWZvbnRTaXplcy1hY2NlbnQtc21fXzF2aHQ0YmpwOiAxMnB4OwogIC0tekluZGV4LWRpYWxvZy1vdmVybGF5X18xdmh0NGJqcTogMTAwMDsKICAtLXpJbmRleC1kaWFsb2ctY29udGVudF9fMXZodDRianI6IDExMDA7CiAgLS16SW5kZXgtZGlhbG9nLXdyYXBwZXJfXzF2aHQ0YmpzOiAxMjAwOwogIC0tdHJhbnNpdGlvbi1wcm9wZXJ0eS1jb21tb25fXzF2aHQ0Ymp0OiBiYWNrZ3JvdW5kLWNvbG9yLCBib3JkZXItY29sb3IsIGNvbG9yLCBmaWxsLCBzdHJva2UsIG9wYWNpdHksIGJveC1zaGFkb3csIHRyYW5zZm9ybTsKICAtLXRyYW5zaXRpb24tZHVyYXRpb24tZmFzdF9fMXZodDRianU6IDAuMXM7CiAgLS10cmFuc2l0aW9uLWR1cmF0aW9uLW5vcm1hbF9fMXZodDRianY6IDAuMnM7CiAgLS1zaGFkb3cteHNfXzF2aHQ0Ymp3OiAwIDAgMCAxcHggcmdiYSgwLCAwLCAwLCAwLjA1KTsKICAtLXNoYWRvdy1zbV9fMXZodDRiang6IDAgMXB4IDJweCAwIHJnYmEoMCwgMCwgMCwgMC4wNSk7CiAgLS1zaGFkb3ctYmFzZV9fMXZodDRiank6IDAgMXB4IDNweCAwIHJnYmEoMCwgMCwgMCwgMC4xKSwgMCAxcHggMnB4IDAgcmdiYSgwLCAwLCAwLCAwLjA2KTsKICAtLXNoYWRvdy1tZF9fMXZodDRiano6IDAgNHB4IDZweCAtMXB4IHJnYmEoMCwgMCwgMCwgMC4xKSwgMCAycHggNHB4IC0xcHggcmdiYSgwLCAwLCAwLCAwLjA2KTsKICAtLXNoYWRvdy1sZ19fMXZodDRiajEwOiAwIDEwcHggMTVweCAtM3B4IHJnYmEoMCwgMCwgMCwgMC4xKSwgMCA0cHggNnB4IC0ycHggcmdiYSgwLCAwLCAwLCAwLjA1KTsKICAtLXNoYWRvdy14bF9fMXZodDRiajExOiAwIDIwcHggMjVweCAtNXB4IHJnYmEoMCwgMCwgMCwgMC4xKSwgMCAxMHB4IDEwcHggLTVweCByZ2JhKDAsIDAsIDAsIDAuMDQpOwogIC0tc2hhZG93LTJ4bF9fMXZodDRiajEyOiAwIDI1cHggNTBweCAtMTJweCByZ2JhKDAsIDAsIDAsIDAuMjUpOwogIC0tc2hhZG93LW91dGxpbmVfXzF2aHQ0YmoxMzogMCAwIDAgM3B4IHJnYmEoNjYsIDE1MywgMjI1LCAwLjYpOwogIC0tc2hhZG93LWlubmVyX18xdmh0NGJqMTQ6IGluc2V0IDAgMnB4IDRweCAwIHJnYmEoMCwwLDAsMC4wNik7CiAgLS1zaGFkb3ctYXJvdW5kX18xdmh0NGJqMTU6IDBweCAwcHggMTVweCByZ2JhKDAsIDAsIDAsIDAuMSk7CiAgLS1zaGFkb3ctbm9uZV9fMXZodDRiajE2OiBub25lOwogIC0tc2hhZG93LWRhcmstbGdfXzF2aHQ0YmoxNzogcmdiYSgwLCAwLCAwLCAwLjEpIDBweCAwcHggMHB4IDFweCwgcmdiYSgwLCAwLCAwLCAwLjIpIDBweCA1cHggMTBweCwgcmdiYSgwLCAwLCAwLCAwLjQpIDBweCAxNXB4IDQwcHg7CiAgLS1ib3JkZXIteHNfXzF2aHQ0YmoxODogOHB4OwogIC0tYm9yZGVyLXNtX18xdmh0NGJqMTk6IDEycHg7CiAgLS1ib3JkZXItbWRfXzF2aHQ0YmoxYTogMTZweDsKICAtLWJvcmRlci1sZ19fMXZodDRiajFiOiAyMHB4OwogIC0tYm9yZGVyLXhsX18xdmh0NGJqMWM6IDI0cHg7CiAgLS1ib3JkZXItZnVsbF9fMXZodDRiajFkOiA5OTk5cHg7CiAgLS1zaXplcy1uYXZiYXItaGVpZ2h0X18xdmh0NGJqMWU6IGNhbGModmFyKC0tc3BhY2UtYmFzZSkgKiAxNSk7CiAgLS1zaXplcy1jb250YWluZXItbmF2YmFyX18xdmh0NGJqMWY6IGNhbGModmFyKC0tc3BhY2UtYmFzZSkgKiAzMjApOwogIC0tc2l6ZXMtY29udGFpbmVyLWNvbnRlbnRfXzF2aHQ0YmoxZzogY2FsYyh2YXIoLS1zcGFjZS1iYXNlKSAqIDI5NSk7Cn0="}!../../../../../../../node_modules/@vanilla-extract/webpack-plugin/extracted/dist/vanilla-extract-webpack-plugin-extracted.cjs.js'
Require stack:
- C:\Users\Surface\Projects\project-name\dist\apps\app-name\.next\server\pages\cro\studies.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\server\require.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\server\load-components.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\build\utils.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\build\output\store.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\build\output\index.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\server\dev\next-dev-server.js
- C:\Users\Surface\Projects\project-name\node_modules\next\dist\server\next.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\next\src\executors\server\lib\custom-server.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\next\src\executors\server\server.impl.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\tao\src\shared\workspace.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\tao\src\commands\run.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\tao\index.js
- C:\Users\Surface\Projects\project-name\node_modules\@nrwl\cli\lib\run-cli.js
```

As mentioned in #396. the issue occurs only for the following import scheme:
```
Component.tsx -> Component.css.ts -> vars.css.ts
```

What's more, the issue (in my case) does not occur on MacOS but only on Windows, which got me thinking if this could be somehow related to the maximum [path length of 260 characters](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd), as my project structure is quite nested. But judging by the fact that #396 was reported on MacOS this could be a false lead.

I've created a [demo project](https://github.com/MrPumpking/vanilla-extract-bug-reproduction) with the structure reflecting the actual project I'm working on (monorepo generated with Nx) which reproduces the bug. The problem occurs when importing and using the theme (`import { theme } from '@demo/ui/theme';`) in `ExampleListItem.css.tss`

```
libs   
├───feature
│   └───domain1
│       └───feature1
│           └───component1
│               └───src
│                   │   ExampleList.tsx
│                   │   index.ts
│                   └───item
│                           ExampleListItem.css.ts
│                           ExampleListItem.tsx
│
└───ui
    └───theme
        └───src
                index.ts
                theme.css.ts
```
Importing the theme in the structure presented above results in the following error:

![image](https://user-images.githubusercontent.com/9134970/151411931-316fa0b8-7f2e-4634-8712-8fe3f5e6ad24.png)

From what I've found moving the `ExampleListItem.css.ts` to the parent directory resolves the problem:

```
libs   
├───feature
│   └───domain1
│       └───feature1
│           └───component1
│               └───src
│                   │   ExampleList.tsx
│                   │   ExampleListItem.css.ts
│                   │   index.ts
│                   └───item
│                           ExampleListItem.tsx
│
└───ui
    └───theme
        └───src
                index.ts
                theme.css.ts
```

![image](https://user-images.githubusercontent.com/9134970/151412539-7a14553a-612c-4a77-a7b6-df46b52a30b6.png)


### Link to reproduction
<!-- Add a link to a repo or codesandbox showcasing the issue. Issues without a reproduction will likely be closed without action. -->

https://github.com/MrPumpking/vanilla-extract-bug-reproduction

There are 2 branches:
- main - contains the code resulting in reported bug
- working-example - working example in which the style file was moved up one directory

### System Info

Output of `npx envinfo --system --npmPackages @vanilla-extract/css,@vanilla-extract/webpack-plugin,@vanilla-extract/esbuild-plugin,@vanilla-extract/vite-plugin,@vanilla-extract/sprinkles,webpack,esbuild,vite --binaries --browsers`:

```node
  System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 6.73 GB / 15.92 GB
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.19.2 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 96.0.4664.45
    Edge: Spartan (44.22000.120.0), Chromium (95.0.1020.53)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @vanilla-extract/css: ^1.6.8 => 1.6.8 
    @vanilla-extract/next-plugin: ^2.0.1 => 2.0.1 
    next: 12.0.7 => 12.0.7 
```

--- a/.changeset/spicy-planes-approve.md
+++ b/.changeset/spicy-planes-approve.md
@@ -1,0 +1,7 @@
+---
+'@vanilla-extract/next-plugin': patch
+---
+
+Fix `Cannot find module *.css.ts.vanilla.css` issue
+
+Previously, CSS was being output on both the client and server builds. This fix ensure CSS is only output on the client build.

--- a/packages/next-plugin/src/index.js
+++ b/packages/next-plugin/src/index.js
@@ -50,7 +50,9 @@ export const createVanillaExtractPlugin =
           ),
         });
 
-        config.plugins.push(new VanillaExtractPlugin(pluginOptions));
+        config.plugins.push(
+          new VanillaExtractPlugin({ outputCss: !isServer, ...pluginOptions }),
+        );
 
         if (typeof nextConfig.webpack === 'function') {
           return nextConfig.webpack(config, options);

